Create board class

JNotelddim opened the pull request.

JNotelddim committed:
Create board class

JNotelddim committed:
Update board property names and create test

mattcasiro commented on src/board.py:
```diff
@@ -0,0 +1,17 @@
+        self._data = data
+
+    @property
+    def snake_list(self):
```
When thinking of how to use this method down the road, I think I prefer a simpler 'snakes' naming. `board.snakes` looks nicer to me than `board.snake_list`. Could do the same with food below for consistency, even though `board.food` could be ambiguous (one food, or multiple foods?) I don't think we'd have trouble understanding it in our context. If we want to suffix with _list because of the data type being an array then we should agree to be consistent with it imo (e.g. I'd change snake.coordinates to snake.coordinate_list). What do you think?

mattcasiro replied:
'Food' for thought, from https://www.python.org/dev/peps/pep-0008/#overriding-principle
> Names that are visible to the user as public parts of the API should follow conventions that reflect usage rather than implementation.

mattcasiro commented on test/board_test.py:
```diff
@@ -0,0 +1,36 @@
+            {
+                "snakes" : snake_list,
+                "food"  : food_list
+            })
```
The closing bracket `)` should be on its own line here for consistency with your other code, though I actually prefer to write this like:
```python
return Board({
    "snakes" : snake_list,
    "food"  : food_list
)}
```

Whichever you prefer works, but I think it looks strange how it is right now. (You could do the same with snake_list if you like that style since it only has one object)
```python
snake_list = [{
    "coords": [{2, 2}, {2,3}, {3,3}],
    "health": 10,
    "id": "asdf1234"
}]
```

I couldn't find anything in the style guide that recommended one way over the other.

mattcasiro commented on src/board.py:
```diff
@@ -0,0 +1,17 @@
+
+    @property
+    def snake_list(self):
+        """snake_list retreives the list of snakes from the board data"""
```
Ok this is being super nit picky, but, I'm sitting at work and they don't have work for me to do. So... you can tell me to lay off, but I'm bored.

I like using an action sentence (not sure if that's the right wording) for comments:
`Get the list of snakes present on the board` instead of `snake_list gets...`

JNotelddim committed:
Update board property naming and docstrings

Use imperative in docstrings to better fit PEP8.
For board list names, replace '_list' suffixes with pluralized variable
names for better intuitiveness.

JNotelddim committed:
Reformat board test indentation

Based on good suggestions from zullybur, just updating the formatting of
indenation on lists for a bit more conistency.

mattcasiro reviewed (approved):
Glorious
